stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/PopUtil.java

```diff
@@ -0,0 +1,210 @@
+            properties.put(ImapConstants.PROPS_IMAP_PORT,
+                    extractPortValue(popConfig.getIntValue(PopConstants.PROPS_PORT), isPop));
+            properties.put(ImapConstants.PROPS_IMAP_AUTH, "true");
+            properties.put(ImapConstants.PROPS_IMAP_STARTTLS, "true");
+            properties.put(EmailConstants.MAIL_STORE_PROTOCOL, ImapConstants.IMAP_PROTOCOL);
+        }
+
+        properties.put(PopConstants.PROPS_USERNAME, popConfig.getStringValue(PopConstants.PROPS_USERNAME));
+        properties.put(PopConstants.PROPS_PASSWORD, popConfig.getStringValue(PopConstants.PROPS_PASSWORD));
+
+        log.debug("[PopUtil][getProperties] Property object created. Returning Object to PopClient");
+        return properties;
+    }
+
+
+    /**
+     * Generates MapValue object using the passed Message.
+     *
+     * @param message received
+     * @return MapValue object
+     * @throws MessagingException If an error occurs related to messaging
+     * @throws IOException If an error occurs related to I/O
+     */
+    public static MapValue getMapValue(Message message) throws MessagingException, IOException {
+        log.debug("[PopUtil][getMapValue] Extracting MapValue values from Config");
+        Map<String, Object> valueMap = new HashMap<>();
+        Address[] toAddress = message.getRecipients(Message.RecipientType.TO);
+        Address[] ccAddress = message.getRecipients(Message.RecipientType.CC);
+        Address[] bccAddress = message.getRecipients(Message.RecipientType.BCC);
+        Address[] replyToAddress = message.getReplyTo();
+        int toAddressArrayLength = getNullArrayLengthChecked(toAddress);
+        int ccAddressArrayLength = getNullArrayLengthChecked(ccAddress);
+        int bccAddressArrayLength = getNullArrayLengthChecked(bccAddress);
+        int replyToAddressArrayLength = getNullArrayLengthChecked(replyToAddress);
+        ArrayValue toAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        ArrayValue ccAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        ArrayValue bccAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        ArrayValue replyToAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        for (int i = 0; i < toAddressArrayLength; i++) {
+            toAddressArrayValue.add(i, toAddress[i].toString());
+        }
+        for (int i = 0; i < ccAddressArrayLength; i++) {
+            ccAddressArrayValue.add(i, ccAddress[i].toString());
+        }
+        for (int i = 0; i < bccAddressArrayLength; i++) {
+            bccAddressArrayValue.add(i, bccAddress[i].toString());
+        }
+        for (int i = 0; i < replyToAddressArrayLength; i++) {
+            replyToAddressArrayValue.add(i, replyToAddress[i].toString());
+        }
+        String subject = getStringNullChecked(message.getSubject());
```

Comment:
We can replace this as follows and remove the `getStringNullChecked` method.
```suggestion
        String subject = message.getSubject() != null ? message.getSubject() : "";
```

Reply:
I think using a method is much cleaner than repeating the same logic.